lib/puppet/configurer.rb

```diff
@@ -224,12 +249,12 @@ def run_internal(options)
     end
 
     begin
+      configured_environment = Puppet[:environment] if Puppet.settings.set_by_config?(:environment)
+
```

Comment:
This got moved because I was originally doing failover logic in this area, but moved it up because it was easier to wrap the bindings around all of run_internal than indent the entire body of this function by another block. I'll move this back to its old location on the next update of this PR